arangod/Aql/DocumentProducingNode.cpp

```diff
@@ -66,9 +67,21 @@ DocumentProducingNode::DocumentProducingNode(ExecutionPlan* plan,
       }
     }
   }
+
+  if (slice.hasKey("filter")) {
+    Ast* ast = plan->getAst();
+    _filter.reset(new Expression(plan, ast, new AstNode(ast, slice.get("filter"))));
```

Comment:
It's purely academic, but if `new Expression()` throws, the new `AstNode` is leaked.

Reply:
No, it doesn't leak. Because `AstNode::AstNode(Ast*, VPackSlice)` will add the AstNode to the Ast's memory management.
So if the constructor returns and `new Expression` fails, it will not leak.
Anyway, I will add a comment there so this is more transparent.

Reply:
Ah good, I didn't know that. Thanks.